cw/cw305/capture.py

```diff
@@ -152,6 +162,107 @@ def capture_aes_random(ot, ktp):
         yield ret
 
 
+def capture_aes_random_batch(ot, ktp, capture_cfg):
+    """A generator for capturing AES traces in batch mode.
+    Fixed key, Random texts.
+
+    Args:
+      ot: Initialized OpenTitan target.
+      ktp: Key and plaintext generator.
+      cfg: Capture configuration.
+    """
+    # Seed host's PRNG.
+    # TODO: Replace this with a dedicated PRNG to avoid other packages breaking our code.
+    random.seed(capture_cfg["batch_prng_seed"])
+    # Set the target's key
+    key = ktp.next_key()
+    tqdm.write(f'Using key: {binascii.b2a_hex(bytes(key))}')
+    cipher = AES.new(bytes(key), AES.MODE_ECB)
+    ot.target.simpleserial_write("k", key)
+    # Seed the target's PRNG 
+    ot.target.simpleserial_write("s", capture_cfg["batch_prng_seed"].to_bytes(4, "little"))
+
+    # Create the ChipWhisperer project.
+    project = cw.create_project(capture_cfg["project_name"], overwrite=True)
+
+    # Capture traces.
+    rem_num_traces = capture_cfg["num_traces"]
+    num_segments_storage = 1
+    # Waverunner scope is not supported. If we need to support it, we can support it not only for 
+    # this function also for other functions as well.
+    scope = create_cw_segmented(ot, capture_cfg)
+    with tqdm(total=rem_num_traces, desc="Capturing", ncols=80, unit=" traces") as pbar:
+        while rem_num_traces > 0:
+            # Determine the number of traces for this batch and arm the oscilloscope.
+            scope.num_segments = min(rem_num_traces, scope.num_segments_max)
+            scope.arm()
+
+            # Start batch encryption.
+            ot.target.simpleserial_write(
+                "b", scope.num_segments_actual.to_bytes(4, "little")
+            )
+
+            # Transfer traces
+            waves = scope.capture_and_transfer_waves()
+            assert waves.shape[0] == scope.num_segments
+
+            # Generate plaintexts and ciphertexts to compare with the batch encryption results.
+            # TODO: In earlier version of the function, scared library is used for AES encryption.
+            # If it does parallel encryption, instead of using Crypto.Cipher it may be used for 
+            # better performance of the script.
+            plaintexts = [ktp.next()[1] for _ in range(scope.num_segments_actual)]
+            ciphertexts = [cipher.encrypt(bytes(plaintexts[ii]))
+                for ii in range(scope.num_segments_actual)]
```

Comment:
In other functions in the same script we are using crypto.cipher and I thought it would be better if we use the same structure. In addition scared library is slower than crypto.cipher. When we try to capture 100k traces with scared we get 1697 traces/s and with crypto.cipher we get 1735 traces/s.